### ISSUE TITLE
adbapi.py:12: 'safe' imported but unused

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3109]
|<img alt="waterbug's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| waterbug reported|
|-|-|
|Trac ID|trac#3109|
|Type|defect|
|Created|2008-03-22 04:11:22Z|
|Branch|https://github.com/twisted/twisted/tree/remove-spurious-warnings-3109|

`twisted.enterprise.adbapi` imports `twisted.enterprise.util.safe` to re-expose it as `twisted.enterprise.adbapi.safe`.  It needs to keep doing this for a while, since `safe` was only recently deprecated.  However, as things are now, just importing `twisted.enterprise.adbapi` (not using `safe` from it) causes a deprecation warning, since all of `twisted.enterprise.util` is deprecated.  If an application only uses undeprecated functionality from adbapi, it should not receive a deprecation warning.

Attachments:

* [adbapi.py.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/96e/96eb91bee768084a554e1397d7a369e3090d8479/9ab09347c65d76f573313b730b2b93fc1e89cabd.patch) (405 bytes) - added by waterbug on 2008-03-22 04:13:08Z - This patch simply removes the import of t.e.utils.safe.
* [patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/96e/96eb91bee768084a554e1397d7a369e3090d8479/d75b3b528276d7a9f30a04b8bccaf74e1d61f67a) (1734 bytes) - added by davep on 2008-03-22 23:56:19Z - remove top-level import but keep 'safe' during deprecation period
* [passing-patch.txt](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/96e/96eb91bee768084a554e1397d7a369e3090d8479/e7a305c5d30608367cc1e9ee40021c56f6064f78.txt) (3216 bytes) - added by radix on 2008-03-23 00:50:46Z - Here's another attempt, this time with working :)  I'm sad that this is so dang complicated!

<details><summary>Searchable metadata</summary>

```
trac-id__3109 3109
type__defect defect
reporter__waterbug waterbug
priority__normal normal
milestone__ 
branch__branches_remove_spurious_warnings_3109 branches/remove-spurious-warnings-3109
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1206159082000000 1206159082000000
changetime__1206235716000000 1206235716000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

[#3022](https://github.com/twisted/twisted/issues/3022) deprecated `twisted.enterprise.util` and everything in it, including `safe`.  Once that's been out for a while, we'll get rid of this stuff.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Apparently I misinterpreted the intent of this ticket.  Adjusting to make it more clear.

**Comment by twisted-trac** [comment]
|[<img alt="jdavisp3's avatar" src="https://avatars.githubusercontent.com/u/43582?s=50" width="50" height="50">](https://github.com/jdavisp3)|@jdavisp3 set owner to @jdavisp3<br>@jdavisp3 set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jdavisp3's avatar" src="https://avatars.githubusercontent.com/u/43582?s=50" width="50" height="50">](https://github.com/jdavisp3)|@jdavisp3 removed owner<br>@jdavisp3 set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set owner to @radix<br>@radix set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix removed owner<br>@radix set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to @radix|
|-|-|

Imports should be at the top of the module, not the middle or inside functions.

Neither `safe` has a docstring now.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_8"></a>|@radix commented|
|-|-|

(In [23021]) Branching to 'remove-spurious-warnings-3109'

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_9"></a>|@radix removed owner|
|-|-|

back up for review in a branch now

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set owner to @radix|
|-|-|

update copyright date on `adbapi.py` and `test_adbapi.py`, then merge.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_11"></a>|@radix set status to `closed`|
|-|-|

(In [23024]) Merge remove-spurious-warnings-3109

Author: davep, radix
Reviewer: radix, exarkun
Fixes [#3109](https://github.com/twisted/twisted/issues/3109)

Importing twisted.enterprise.adbapi no longer warns the user spuriously
about twisted.enterprise.util being deprecated. Now adbapi.safe has
a separate deprecation warning.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
